internal/service/alarms/internal/db/models/alarm_event_record.go

```diff
@@ -3,30 +3,35 @@ package models
 import (
 	"time"
 
+	"github.com/stephenafamo/bob/dialect/psql"
+
 	"github.com/google/uuid"
+	"github.com/openshift-kni/oran-o2ims/internal/service/alarms/api/generated"
 )
 
 // AlarmEventRecord represents a record in the alarm_event_record table.
 type AlarmEventRecord struct {
-	AlarmEventRecordID    uuid.UUID              `db:"alarm_event_record_id"`
-	AlarmDefinitionID     uuid.UUID              `db:"alarm_definition_id"`
-	ProbableCauseID       uuid.UUID              `db:"probable_cause_id"`
-	AlarmRaisedTime       time.Time              `db:"alarm_raised_time"`
-	AlarmChangedTime      *time.Time             `db:"alarm_changed_time"`
-	AlarmClearedTime      *time.Time             `db:"alarm_cleared_time"`
-	AlarmAcknowledgedTime *time.Time             `db:"alarm_acknowledged_time"`
-	AlarmAcknowledged     bool                   `db:"alarm_acknowledged"`
-	PerceivedSeverity     int                    `db:"perceived_severity"`
-	Extensions            map[string]interface{} `db:"extensions"`
-	ResourceID            uuid.UUID              `db:"resource_id"`
-	ResourceTypeID        uuid.UUID              `db:"resource_type_id"`
-	NotificationEventType int                    `db:"notification_event_type"`
-	AlarmStatus           string                 `db:"alarm_status"`
-	Fingerprint           string                 `db:"fingerprint"`
-	AlarmSequenceNumber   int64                  `db:"alarm_sequence_number"`
-	CreatedAt             time.Time              `db:"created_at"`
+	AlarmEventRecordID    *uuid.UUID                  `db:"alarm_event_record_id"`
```

Comment:
Why is this being made nullable?  It is the primary key so it should always be there.    Is it because we are relying on the DB to set it on the insert?

Reply:
> Is it because we are relying on the DB to set it on the insert

Yeah exactly. Normally when inserting the live data...we are auto generating uuid with db. but when ready to move to archive it will already have a UUID. This var is basically used to determine if we working with archive or live table and dynamically update the queries.